public/js/app/hooks/usePlannedModules.js

```diff
@@ -12,21 +12,23 @@ export const usePlannedModules = ({ courseCode, semester, applicationCode }) =>
 
   const basePath = context.paths.api.plannedSchemaModules.uri
 
-  const { data, isError, setApiParams } = useApi(
-    getPlannedModules,
-    { basePath, courseCode, semester, applicationCode },
-    null,
-    MISSING_INFO
+  const requestData = useMemo(
+    () => ({
+      basePath,
+      courseCode,
+      semester,
```

Comment:
I found a bug, the semester prop in the params object is called `selectedSemester` and not `semester`, so semester is always undefined here.

Reply:
This still doesn't work for when I'm testing...

Reply:
Ah, it is still called semester in the planned modules api function. So it needs to be `semester: selectedSemester,` here.

Like this: 
```
  const requestData = useMemo(
    () => ({
      basePath,
      courseCode,
      semester: selectedSemester,
      applicationCode,
    }),
    [basePath, courseCode, selectedSemester, applicationCode]
  )
```

Reply:
You are absolutely right, I fixed it now; thanks for mentioning this.